src/main/java/com/linkedin/kmf/services/ClusterTopicManipulationService.java

```diff
@@ -171,7 +175,28 @@ private boolean doesClusterContainTopic(String topic, Collection<Node> brokers,
 
       totalPartitionsInCluster += this.processBroker(logDirectoriesResponseMap, broker, topic);
     }
-    return totalPartitionsInCluster == expected;
+    return totalPartitionsInCluster == expected && ClusterTopicManipulationService.isTopicDescribeSuccessful(
+        adminClient, Collections.singleton(topic));
+  }
+
+  /**
+   * Waits if necessary for this future to complete, and then returns its result.
+   * returns true if the future succeeds, which occurs only if all the topic descriptions are successful.
+   * @param adminClient administrative client for Kafka, supporting managing and inspecting topics, brokers, configurations and ACLs.
+   * @param topicNames Collection of topic names
+   * @return boolean value if describe topic succeeds.
+   */
+  private static boolean isTopicDescribeSuccessful(AdminClient adminClient, Collection<String> topicNames) {
```

Comment:
This function does 2 things, waiting for the describe PRC to finish, check whether it failed and return a boolean. I would suggest the following:

```
private static Map<String, TopicDescription> describeTopics(AdminClient adminClient, Collection<String> topicNames) throws  InterruptedException, ExecutionException {
   KafkaFuture<Map<String, TopicDescription>> mapKafkaFuture = adminClient.describeTopics(topicNames).all();
   return mapKafkaFuture.get();
}
```

Then make the caller handle exception if there is any and decide whether topic description is successful or not. This way is cleaner since the function is doing only one thing instead of two things. You can also have the option to do something with the returned value if it's necessary in the future.

Reply:
ack.